### PR TITLE
[WNMGDS-2290] Upgrade `downshift` and convert `Autocomplete` to functional component

### PR DESCRIPTION
**Pull request by pwolfert**
## Summary

https://jira.cms.gov/browse/WNMGDS-2290

This is the 1st of 2 pull requests. The `downshift` team has kept its older API around, so I did the upgrade without really changing our `downshift` implementation, and then I converted it to a functional component in https://github.com/CMSgov/design-system/commit/befe75162f67dcc26f29ceec78da208d58d6d82c to prepare for the next step of using their new Hooks API.

## How to test

Test out the Autocomplete. Should be no bugs or functional differences.

## Checklist

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title` or [NO-TICKET] if this is unticketed work.
- [x] Selected appropriate `Type` (only one) label for this PR, if it is a breaking change, label should only be `Type: Breaking`
- [x] Selected appropriate `Impacts`, multiple can be selected.

### CODE REVIEW

File: packages/design-system/src/components/Autocomplete/Autocomplete.tsx [hunk 1]
```diff
@@ -161,42 +166,42 @@ function isTextField(child: React.ReactElement): boolean {
...
+    ariaClearLabel,
+    autoCompleteLabel,
+    autoFocus,
+    children,
+    className,
+    clearInputOnBlur,
+    clearInputText,
+    clearSearchButton,
+    focusTrigger,
+    id: _id,
```
Comment (forrestbaer) [inline packages/design-system/src/components/Autocomplete/Autocomplete.tsx:186]: I'm a bit confused by how this destructuring assignment works. `_id` isn't used anywhere else and I don't see it defined. Is this some react magic that I'm not grasping? The only mention of using underscore prefixes was in a `don't do` document relating to using underscores to faux mock private methods as a design choice (https://github.com/airbnb/javascript/tree/master/react#methods). 

The outcome of this is that it seems like you're storing a copy of id in `_id` which isn't used anywhere else., same for `_labelId`, curious how this works.

**Review by forrestbaer — APPROVED** [review]
This looks good, analytics working exactly the same, no functional differences that I can spot, just one question regarding destructuring of a few items.

**Comment by forrestbaer** [comment]
Also is this the correct milestone, this is going with 7? I've set up another milestone for 8.0 if that's where it should go instead?

File: packages/design-system/src/components/Autocomplete/Autocomplete.tsx (hunk 1 above)
Comment (pwolfert) [reply packages/design-system/src/components/Autocomplete/Autocomplete.tsx:186]: The reason I'm including it in the destructuring assignment is to pull it out of the remainder (`...autocompleteProps`), and I'm underscoring it to signal that we don't use it and so it doesn't conflict with the `id` and `labelId` defined at the top of this function.

**Pull request merged**